MobiusCore/Source/EffectHandlers/EffectCallback.swift

```diff
@@ -55,16 +54,14 @@ public final class EffectCallback<Output> {
     }
 
     public func send(_ output: Output) {
-        lock.synchronized {
```

Comment:
We cannot know how long this lock will be held. Calls to the output consumer are serialized by the framework anyway. The only thing we need to ensure here is that `end` is not called while the output consumer is being used.